products/BellHybrid/services/db/tests/MeditationStatisticsTable_tests.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include <catch2/catch.hpp>
+
+#include <MeditationStatisticsDB.hpp>
+
+#include <filesystem>
+#include <iostream>
+#include <gsl/util>
+
+namespace
+{
+    using namespace db::meditation_stats;
+
+    time_t subtract_time(std::chrono::minutes minutes)
+    {
+        return std::chrono::system_clock::to_time_t(std::chrono::system_clock::now() - minutes);
+    }
+
+    time_t get_utc_time()
+    {
+        return std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+    }
+
+    template <typename Db> class TestDatabase
+    {
+      public:
+        explicit TestDatabase(std::filesystem::path name)
+        {
+            Database::initialize();
+
+            if (std::filesystem::exists(name)) {
+                REQUIRE(std::filesystem::remove(name));
+            }
+
+            db = std::make_unique<MeditationStatisticsDB>(name.c_str());
+
+            if (not db->isInitialized()) {
+                throw std::runtime_error("Could not initialize database");
+            }
+        }
+
+        ~TestDatabase()
+        {
+            Database::deinitialize();
```

Comment:
would not be better to delete database after test is done ?

Reply:
I decided it was handy to have a database file in case of unit tests error. We would not have access to it otherwise.